tests/migrations/test_update_legacy_filtered_positives.py

```diff
@@ -0,0 +1,193 @@
+
+        mock_update_mongo.assert_not_called()
+        mock_update_mlwh.assert_not_called()
+
+
+def test_update_legacy_filtered_positives_catches_error_connecting_to_mongo(
+    mock_helper_database_updates, mock_v0_version_set
+):
+    with pytest.raises(Exception):
+        with patch(
+            "migrations.update_legacy_filtered_positives.legacy_mongo_samples",
+            side_effect=Exception("Boom!"),
+        ):
+            mock_v0_version_set.return_value = False
+            mock_update_mongo, mock_update_mlwh = mock_helper_database_updates
+            update_legacy_filtered_positives.run("crawler.config.integration")
+
+            mock_update_mongo.assert_not_called()
+            mock_update_mlwh.assert_not_called()
+
+
+def test_error_connecting_to_mysql_databases_raises_exception(
+    mock_v0_version_set,
+    mock_helper_database_updates,
+    mock_query_helper_functions,
+    mock_extract_required_cp_info,
+):
+    mock_update_mongo, mock_update_mlwh = mock_helper_database_updates
+    mock_legacy_mongo_samples, mock_get_cherrypicked_samples_by_date = mock_query_helper_functions
+
+    mock_v0_version_set.return_value = False
+    mock_legacy_mongo_samples.return_value = [{"plate_barcode": "1"}]
+    mock_extract_required_cp_info.return_value = [["id_1"], ["plate_barcode_1"]]
+    mock_get_cherrypicked_samples_by_date.side_effect = Exception("Boom!")
+
+    with pytest.raises(Exception):
+        update_legacy_filtered_positives.run("crawler.config.integration")
+        mock_update_mongo.assert_not_called()
+        mock_update_mlwh.assert_not_called()
+
+
+def test_update_legacy_filtered_positives_outputs_success(
+    mock_v0_version_set,
+    mock_helper_database_updates,
+    mock_query_helper_functions,
+    mock_extract_required_cp_info,
+    mock_helper_functions,
+    mock_update_filtered_positive_fields,
+):
+    mock_update_mongo, mock_update_mlwh = mock_helper_database_updates
+    mock_legacy_mongo_samples, mock_get_cherrypicked_samples_by_date = mock_query_helper_functions
+    mock_split_mongo_samples_by_version = mock_helper_functions
+
+    mock_v0_version_set.return_value = False
+    mock_legacy_mongo_samples.return_value = [{"plate_barcode": "1"}]
+    mock_extract_required_cp_info.return_value = [["id_1"], ["plate_barcode_1"]]
+    mock_get_cherrypicked_samples_by_date.return_value = pd.DataFrame({"id": ["s1", "s2"]})
+    mock_split_mongo_samples_by_version.return_value = {
+        FilteredPositiveIdentifierV0(): [{"plate_barcode": "1"}],
+        FilteredPositiveIdentifierV1(): [{"plate_barcode": "2"}],
+        FilteredPositiveIdentifierV2(): [{"plate_barcode": "3"}],
+    }
+    mock_update_mongo.return_value = True
+    mock_update_mlwh.return_value = True
+
+    update_legacy_filtered_positives.run("crawler.config.integration")
+
+    assert mock_update_filtered_positive_fields.call_count == 3
+    assert mock_update_mongo.call_count == 3
+    assert mock_update_mlwh.call_count == 3
```

Comment:
Worth also checking what arguments these methods were called with, e.g. the correct list of samples?

Also in below test in which user input is required

Reply:
👍  good point, done this